src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -795,6 +850,13 @@ public async Task<HttpResponseMessage> SendAsyncCore(HttpRequestMessage request,
                 // Make sure to complete the allowExpect100ToContinue task if it exists.
                 allowExpect100ToContinue?.TrySetResult(false);
 
+                if (_readAheadTask != default)
```

Comment:
Do we care about distinguishing between default and a read-ahead that synchronously completed with 0?

Reply:
I don't think so. At least for comparisons within the request itself (SendAsync, FillAsync ...), we should never see a sync-completing read-ahead task. If it were to complete synchronously, it should have immediately claimed completion ownership, and we therefore shouldn't have started sending the request on the connection.

Reply:
Ok.  Since a synchronously completing operation that returns 0 is represented identically to default, it'd be worth at least a comment somewhere as to why these comparisons are ok.